[ATSPI] Modal element containing only StaticText is ignored

obyknovenius opened the pull request:
#### 99e0cc3c3dd2a926cd667cfab52cf403a0bfbae8
<pre>
[ATSPI] Modal element containing only StaticText is ignored
<a href="https://bugs.webkit.org/show_bug.cgi?id=246329">https://bugs.webkit.org/show_bug.cgi?id=246329</a>

Reviewed by Carlos Garcia Campos.

A modal element will not be set as the current modal when it has no
accessible content. But when using AT-SPI, an object with
AccessibilityRole::StaticText is ignored, since we expect its parent
to implement org.a11y.atspi.Text interface itself.

Fixes `accessibility/custom-elements/modal.html` time out.

* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::modalElementHasAccessibleContent):

Canonical link: <a href="https://commits.webkit.org/257235@main">https://commits.webkit.org/257235@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ab452d350090b1536cef90596f0334ba317595f9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/92385 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/1614 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/22979 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/102153 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/162637 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/96386 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/1610 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/29995 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/84816 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/98311 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/98046 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/1066 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/78899 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/28005 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/82984 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/82677 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/71103 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/36411 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/16619 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/34178 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/17789 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4982 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/38049 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/40404 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39950 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/36939 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebCore/accessibility/AXObjectCache.cpp:
```diff
@@ -299,6 +299,11 @@ bool AXObjectCache::modalElementHasAccessibleContent(Element& element)
+#if USE(ATSPI)
+                if (axObject->roleValue() == AccessibilityRole::StaticText)
+                    return true;
+#endif
```
Add a comment here explaining why this is different for ATSPI, please.

obyknovenius committed:
[ATSPI] Modal element containing only StaticText is ignored
https://bugs.webkit.org/show_bug.cgi?id=246329

Reviewed by Carlos Garcia Campos.

A modal element will not be set as the current modal when it has no
accessible content. But when using AT-SPI, an object with
AccessibilityRole::StaticText is ignored, since we expect its parent
to implement org.a11y.atspi.Text interface itself.

Fixes `accessibility/custom-elements/modal.html` time out.

* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::modalElementHasAccessibleContent):

Canonical link: https://commits.webkit.org/257235@main
